nextjs/components/EmployeeView.tsx

```diff
@@ -1,36 +1,76 @@
 import { useAuth } from '@/context/AuthContext'
-import { FullUserDTO } from '@/types'
+import { FullUserDTO, WorkHourDTO } from '@/types'
 import { env } from 'process'
 import React, { useEffect, useState } from 'react'
 import Calendar from 'react-calendar'
 import 'react-calendar/dist/Calendar.css'
 import CircularButton from './CircularButton'
-
-interface WorkHoursData {
-  [date: string]: number
-}
+import { format } from 'date-fns'
+import styles from '@/styles/employeeview.module.css'
 
 function EmployeeView() {
-  const [date, setDate] = useState<Date>(new Date())
   const { user } = useAuth()
-  const [workHoursData, setWorkHoursData] = useState<WorkHoursData>({})
   const [dbUser, setDbUser] = useState<FullUserDTO | undefined>(undefined)
-  const [selectedUser, setSelectedUser] = useState<FullUserDTO | undefined>()
   useEffect(() => {
     // Fetch the list of users from your database API and update the state
     // Replace the URL with your actual API endpoint
     fetch(`/api/user?username=${user?.preferred_username}`)
       .then((response) => response.json())
       .then((data) => setDbUser(data))
       .catch((error) => console.error(error))
-  }, [])
+  }, [user])
 
+  if (!dbUser) {
+    return
+  }
   const currentDate = new Date()
 
+  const onClick = () => {
+    const newDbUser: FullUserDTO = { ...dbUser }
+    newDbUser.workHours = undefined
+    if (dbUser.isWorking && dbUser.startAt) {
+      newDbUser.isWorking = false
+      newDbUser.startAt = null
+      const newWorkHour: WorkHourDTO = {
+        startTime: dbUser.startAt,
+        endTime: currentDate,
+        userId: dbUser.id,
+        date: format(currentDate, 'yyyy-MM-dd HH:mm:ss'),
+      }
+      fetch('/api/workhour', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(newWorkHour),
+      })
+    } else {
+      newDbUser.startAt = currentDate
+      newDbUser.isWorking = true
+    }
+    setDbUser(newDbUser)
+    fetch('/api/user', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(newDbUser),
+    })
+  }
+  console.log(dbUser.startAt)
```

Comment:
Consider using a linter so that console.logs are not merged to production.  Any message leaking to console on production should be intentional, IMO.